### PR TITLE
Don't explicitly free the C4Document, when freeing the Document 

### PR DESCRIPTION
**Pull request by bmeike**
Beautify one more class
 CBL-601: Don't explicitly free the C4Document, when freeing the Document

### CODE REVIEW

**Comment by bmeike** [comment]
Also explored making Document thread-safe.

File: lib/src/shared/main/java/com/couchbase/lite/Document.java [hunk 1]
```diff
@@ -523,19 +535,6 @@ final FLSliceResult encode() throws LiteCoreException {
...
-        root = null;
-
-        final C4Document c4Doc;
-        synchronized (lock) {
-            c4Doc = c4Document;
-            c4Document = null;
-        }
-
-        // c4doc should be retained.
-        if (c4Doc != null) { c4Doc.release(); }
```
Comment (pasin) [inline lib/src/shared/main/java/com/couchbase/lite/Document.java]: This is getting a little messy when retain and release are not matched. In updatedC4DocumentLocked(C4Document) we retain the c4Doc but not there is no where to call the release(). I tried to search where the retain() and release() method are called but I couldn't see if it's necessary to call retain()/release() at all as the C4Document will be released by itself. 

@bmeike Have you tried to clean this up, maybe by removing retain()/release() calls?

Comment (bmeike) [reply lib/src/shared/main/java/com/couchbase/lite/Document.java]: I have tried to clean it up... but not in a single commit.  This is already ugly.
Let look at this again.  I may be able to just remove them all, altogether, as you suggest.

**Comment by bmeike** [comment]
This PR is on hold until integration with the new Core is complete

**Comment by bmeike** [comment]
Abandoning this PR until after the core update

**Pull request closed without merging**